packages/react/src/useFetch.ts

```diff
@@ -122,22 +145,55 @@ export function useFetch<T = string>(path: string, options?: FetchHookOptions):
           throw new Error(response.statusText);
         }
 
+        let dispatchData = true;
+
         if (mergedOptions.json) {
           data = await response.json();
+        } else if (typeof mergedOptions.stream === "string") {
+          dispatchData = false;
+          const decodedStream = response.body!.pipeThrough(
+            new TextDecoderStream(mergedOptions.stream === "string" ? "utf8" : mergedOptions.stream)
+          );
+
+          const [responseStream, updateStream] = decodedStream.tee();
+
+          data = responseStream;
+          const decodedStreamReader = updateStream.getReader();
+
+          decodedStreamReader.closed.catch((error) => {
+            dispatchError(mounted, dispatch, error, response);
+          });
+
+          dispatch({ type: "fetched", payload: "" as any });
+
+          (async () => {
+            let responseText = "";
+            let done = false;
+
+            while (!done) {
+              const { value, done: _done } = await decodedStreamReader.read();
+              done = _done;
+
+              if (value) {
+                responseText += value;
+                dispatch({ type: "update", payload: responseText as any });
+              }
+            }
+          })().catch((error) => {
+            dispatchError(mounted, dispatch, error, response);
+          });
```

Comment:
good call; I've added in something that will make sure that this doesn't do funky stuff across multiple sends

I'm not super confident on if we want to handle aborting the first request or if we should leave it to the user to do that; thoughts @thegedge @airhorns?

Reply:
I think we should abort the first request -- it's response isn't going to go anywhere. That's also what browsers do on navigation, if you have one going and leave the page, the browser kills it. Since it might cost tokens I think that's wise as well!